content/en/blog/2022/opamp/index.md

```diff
@@ -0,0 +1,239 @@
+   }
+   return http.HandlerFunc(httpHandler)
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   exp, _ := newConsoleExporter()
+   tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exp))
+   defer func() { _ = tp.Shutdown(context.Background()) }()
+
+   otel.SetTracerProvider(tp)
+
+   tracer = tp.Tracer("ControlledOpAMPAgentDemo")
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", setHandler(handler, config))
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Build it with `go build .` Now if we run it and open http://localhost:8080/ on a
+browser, nothing special will be shown. It’s time to add some opamp. Git clone
+[opamp-go](https://github.com/open-telemetry/opamp-go) and run the server with:
+
+```
+cd internal/examples/server
+go run .
+```
+
+Go to http://localhost:4321/ to verify it running, notice no agent displayed:
+
+![No agents display on opamp server demo UI](opamp_server_no_agents.png)
+
+Next, open a new terminal and on internal/examples/supervisor/bin edit
+supervisor.yaml agent executable to point to our previous build. On
+internal/examples/supervisor run `go build -o bin/supervisor main.go` and on on
+internal/examples/supervisor/bin run `./supervisor`
+
+Great, we have now a system consisting of OpAMP server supervisor and our server
+
+![OpAMP server, supervisor and agent relations](opamp_server_supervisor_agent_relations.png)
+
+We can see now our agent running at http://localhost:4321/. Select it and pass
+`instrument: true` to its configurations.
+
+![Our service configurations over opamp server](opamp_server_agent_config.png)
+
+We can see the change take place on the supervisor console log:
+
+```
+Received remote config from server, hash=0008886301f3ccb3520216823cfa09a.
+Effective config changed.
+Config is changed. Signal to restart the agent.
+Restarting the agent with the new config.
+Stopping agent process, PID=19206
+Agent process PID=19206 successfully stopped.
+Starting agent <agent path>
+Agent process started, PID=19506
+```
+
+Finally, visiting http://localhost:8080/ should result with traces appear on
+agent.log in internal/examples/supervisor/bin:
+
+```go
+Starting server on port8080
+{
+   "Name": "instrumentation activated by OpAMP server",
+   "SpanContext": {
+      "TraceID": "d2f76958023624d4c1def3f44899b6d4",
+      "SpanID": "085510f551dc31a1",
+      "TraceFlags": "01",
+      "TraceState": "",
+      "Remote":false
+   ...
+```
+
+The green lines are the trace itself! Ok, so what do we have here? A server that
+controls whether or not our service will generate traces. Try setting it off
+using the `instrument: false` configuration.
+
+Ok, so what’s next? This is the very basic implementation, wrapping a system
```

Comment:
"What's next?"

Reply:
fixed via the [other suggestion below ](https://github.com/open-telemetry/opentelemetry.io/pull/1847#discussion_r992842915)